frontend/src/metabase/databases/components/DatabaseForm/field-grouping.unit.spec.ts

```diff
@@ -0,0 +1,138 @@
+import type { EngineField, FieldGroupConfig } from "metabase-types/api";
+
+import { GroupedFields, groupFields } from "./field-grouping";
+
+describe("groupFields", () => {
+  const hostField: EngineField = {
+    name: "host",
+    type: "string",
+    "group-id": "host-port",
+  };
+
+  const portField: EngineField = {
+    name: "port",
+    type: "integer",
+    "group-id": "host-port",
+  };
+
+  const databaseField: EngineField = {
+    name: "database",
+    type: "string",
+  };
+
+  const fieldGroupConfig: FieldGroupConfig = {
+    id: "host-port",
+    "container-style": "flex",
```

Comment:
maybe leave room from an options param to be consistent with the grid formatting 

```suggestion
    "container-style": ["flex"],
```